Format man page

Honny1 opened the pull request:
This PR improves the format of the man page.

Honny1 commented:
/packit copr-build

Honny1 commented:
/packit build

jan-cerny commented on openscap_report/cli.py:
```diff
@@ -18,23 +18,23 @@
-    "LOG LEVELS:\n"
-    "\tDEBUG - Detailed information, typically of interest only for diagnosing problems.\n"
-    "\tINFO - A confirmation that things are working as expected.\n"
+    "\nLOG LEVELS:\n"
```
For these types of strings where you want to preserve whitespace etc. a triple-quoted string `""" ... """` could work better.

jan-cerny commented on openscap_report/cli.py:
```diff
@@ -18,23 +18,23 @@
+    "\tDEBUG - Detailed information, typically of interest only for diagnosing problems.\n\n"
+    "\tINFO - A confirmation that things are working as expected.\n\n"
     "\tWARING -  An indication that something unexpected happened, or a signal of"
-    " a possible problem in the future. The software is still working as expected.\n"
+    " a possible problem in the future. The software is still working as expected.\n\n"
```
WARNING

jan-cerny commented on openscap_report/cli.py:
```diff
@@ -85,7 +101,7 @@ def prepare_parser():
         choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
         help=(
-            "write debug information to the log up to the LOG_LEVEL."
+            "write debug information to the log up to the LOG_LEVEL. (default: %(default)s)"
```
This default section causes that there is the default described twice in the man page.

jan-cerny commented on openscap_report/cli.py:
```diff
@@ -43,10 +43,26 @@
+        else:
+            default = action.dest.upper()
+            args_string = self._format_args(action, default)
+            parts.append(f'{", ".join(action.option_strings)} {args_string}')
```
There is too much braces and quotes on this line, which makes it harder to read. I think it would be more readable if the `", ".join(action.option_strings)` part would be extracted to a variable.

jan-cerny reviewed (changes requested):
It's a great improvement.

But, the generated man page has some problems with formatting of command line options. For example it shows `d, --debug` instead of `-d, --debug`. Also, the `FILE` positional argument isn't visible.

pep8speaks commented:
Hello @Honny1! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`openscap_report/cli.py`](https://github.com/OpenSCAP/openscap-report/blob/57035a4f2c8ead066d1a62bea40d811f96d0a7d8/openscap_report/cli.py):

> [Line 27:100](https://github.com/OpenSCAP/openscap-report/blob/57035a4f2c8ead066d1a62bea40d811f96d0a7d8/openscap_report/cli.py#L27): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (156 > 99 characters)
> [Line 29:100](https://github.com/OpenSCAP/openscap-report/blob/57035a4f2c8ead066d1a62bea40d811f96d0a7d8/openscap_report/cli.py#L29): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (118 > 99 characters)
> [Line 31:100](https://github.com/OpenSCAP/openscap-report/blob/57035a4f2c8ead066d1a62bea40d811f96d0a7d8/openscap_report/cli.py#L31): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 99 characters)
> [Line 43:100](https://github.com/OpenSCAP/openscap-report/blob/57035a4f2c8ead066d1a62bea40d811f96d0a7d8/openscap_report/cli.py#L43): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (133 > 99 characters)

jan-cerny commented:
@Honny1 What has changed? Have you addressed the feedback?

Honny1 commented:
@jan-cerny Yes, I have addressed the feedback. I am currently trying to replicate the problem you described in the first place. But I can't reproduce the problem.

Honny1 commented:
@jan-cerny I have successfully reproduced the problem with the formatting of command line options. It looks like this is caused by some bug in the old version of the `python3-sphinx` (`4.1.2`) package. I'm using `python3-sphinx` (`4.4.0`) and it looks fine. What we can do with this problem?

Honny1 commented:
@jan-cerny I found the main cause of the problem with the formatting of command line options. In generated man page are missed white spaces before the parameter texts. That's why `man` does not show the first dash or `FILE` text. 

Here is part of the diff in the man pages generated by the different versions of the packages: 
```
95c97
< .B \-d, \-\-debug
---
> .B\-d, \-\-debug
```
I am currently working on a fix.

Honny1 commented:
@jan-cerny The fix is uploaded. You can test it.

jan-cerny commented:
Yes, it could be caused by the version. I'm using python3-sphinx-4.1.2-3.fc35.noarch. But, your fix didn't help, I still have the issue with missing dashes in options in the generated man page. It still generates  lines like `.B\-d, \-\-debug`.

Honny1 commented:
@jan-cerny  How did you generate a man page? The default behaviour of `sphinx-build` is to only read and parse source files that are new or have changed since the last run. My fix changed the way how arguments are parsed. 
So if you just run `sphinx-build -b man . TARGET_DIR`. Nothing should be changed on the generated man page. I suggest you regenerate the man page with this command `sphinx-build -b man -E . TARGET_DIR`.

jan-cerny commented:
@Honny1 Great! That was it! I was simply expecting that it overwrites existing files. If I do some cleaning and remove them before, it works fine.
